build-system/pr-check/build-targets.js

```diff
@@ -214,14 +243,16 @@ const targetMatchers = {
 };
 
 /**
- * Populates buildTargets with a set of build targets contained in a PR after
- * making sure they are valid. Used to determine which checks to perform / tests
- * to run during PR builds.
- * @return {boolean}
+ * Returns the set of build targets affected by a PR after making sure they are
+ * valid. Used to determine which checks to perform / tests to run during PR
+ * builds. Exits early if targets have already been populated.
+ * @return {Set<string>}
  */
 function determineBuildTargets() {
+  if (buildTargets.size > 0) {
+    return buildTargets;
+  }
```

Comment:
Technically it's possible that even after calculating this buildTargets none are found (not likely because that would mean a meaningless commit, but it is possible) - how about changing the default value of buildTargets to `undefined` and assign it to `new Set()` inside this function, replacing the cache-check with `if (buildTargets !== undefined)`

Reply:
> Technically it's possible that even after calculating this buildTargets none are found

We already have a way to prevent this. If a file matches no known build targets, it defaults to `RUNTIME`. (So, if someone checks in a new kind of file for which there isn't an existing build target, we assume it affects the runtime and run a whole slew of tests on it.)

https://github.com/ampproject/amphtml/blob/cb907bee87bf1de15654e3afd6c899ba6815dde3/build-system/pr-check/build-targets.js#L234-L236